packages/react-mutation-mapper/src/store/DefaultMutationMapperStore.ts

```diff
@@ -92,50 +91,72 @@ class DefaultMutationMapperStore implements MutationMapperStore {
     @observable
     private _activeTranscript: string | undefined = undefined;
 
+    @observable
+    private _selectedTranscript: string | undefined = undefined;
+
     constructor(
         public gene: Gene,
         protected config: DefaultMutationMapperStoreConfig,
         protected getMutations: () => Mutation[]
     ) {}
 
-    @computed
-    public get activeTranscript(): string | undefined {
-        let activeTranscript;
-        const canonicalTranscriptId =
-            !this.canonicalTranscript.isPending &&
-            this.canonicalTranscript.result
-                ? this.canonicalTranscript.result.transcriptId
-                : undefined;
-
-        if (this._activeTranscript !== undefined) {
-            activeTranscript = this._activeTranscript;
-        } else if (this.config.filterMutationsBySelectedTranscript) {
-            if (
-                this.transcriptsWithAnnotations.result &&
-                this.transcriptsWithAnnotations.result.length > 0 &&
-                canonicalTranscriptId &&
-                !this.transcriptsWithAnnotations.result.includes(
-                    canonicalTranscriptId
-                )
-            ) {
-                // if there are annotated transcripts and activeTranscript does
-                // not have any, change the active transcript
-                activeTranscript = this.transcriptsWithAnnotations.result[0];
-            } else {
-                activeTranscript = canonicalTranscriptId;
-            }
-        } else {
-            activeTranscript = canonicalTranscriptId;
-        }
+    readonly activeTranscript: MobxPromise<string | undefined> = remoteData(
+        {
+            await: () => [this.allTranscripts, this.canonicalTranscript],
+            invoke: async () => {
+                let activeTranscript;
+                const canonicalTranscriptId = this.canonicalTranscript.result
+                    ? this.canonicalTranscript.result.transcriptId
```

Comment:
since you await it above, you can just reference result.transcriptId without checking.   if TS gives you an error, you can use ```this.canonicalTranscript.result!``` which tells TS you KNOW it will be !undefined

Reply:
same as `allTranscripts`, the`canonicalTranscript` could be undefined sometimes, probably I need to keep this to avoid breaking something?